### PR TITLE
glsl_shader_gen: Remove invariant qualifier

### PR DESCRIPTION
**Pull request by GPUCode**
Seems to be the cause of the depth glitches in RADV. Also made sure the null surface is cleared to transparent as RADV also exhibited the black squares in the player's face

Citra (master) | Citra (PR)
-|-
Pokémon Y_21 01 24_12 29 27 775 | Pokémon Y_21 01 24_12 28 36 403

Fixes:
https://github.com/citra-emu/citra/issues/7197

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7376)

### CODE REVIEW

**Comment by GPUCode** [comment]
@Steveice10 Can you check if the same problem comes back with MoltenVK? We may need to apply this only for that in that case. Though it's pretty strange that it fixes one platform and breaks another

**Comment by Steveice10** [comment]
> Can you check if the same problem comes back with MoltenVK? We may need to apply this only for that in that case. Though it's pretty strange that it fixes one platform and breaks another

Yes, this breaks on MoltenVK because Apple GPUs apply aggressive shader optimizations where allowed that result in too much variance. I don't see why adding invariant should break this effect given how it works, if anything I would think we should be disabling invariant only on RADV as it sounds like a RADV issue.

**Comment by GPUCode** [comment]
The qualifier doesn't appear to be often used by client applications so I agree this might be a RADV bug. However I'm in favor of enabling this in cases where it's needed instead of always enabling. Most drivers except for MoltenVK are fine without it, and it's easier to just apply it on all apple platforms compared to detecting buggy RADV versions.

**Comment by Steveice10** [comment]
> The qualifier doesn't appear to be often used by client applications so I agree this might be a RADV bug. However I'm in favor of enabling this in cases where it's needed instead of always enabling. Most drivers except for MoltenVK are fine without it, and it's easier to just apply it on all apple platforms compared to detecting buggy RADV versions.

I guess I'm not strongly opposed to it, but this basically means if any driver wants to apply optimizations in the future that introduce variance we have nothing telling the drivers not to.

**Comment by GPUCode** [comment]
Given the recognizable pattern of glitching this variance exhibits and the fact it occurs only in this game it should be pretty trivial to apply this for more vendors in the future. If it starts occurring in other vendors we can start enabling it globally again

File: src/video_core/shader/generator/glsl_shader_gen.cpp [hunk 1]
```diff
@@ -63,7 +63,11 @@ static std::string GetVertexInterfaceDeclaration(bool is_output, bool use_clip_p
     if (is_output && separable_shader) {
         // gl_PerVertex redeclaration is required for separate shader object
         out += "out gl_PerVertex {\n";
+#ifdef __APPLE__
+        out += "    invariant vec4 gl_Position;\n";
```
Comment (Steveice10) [inline src/video_core/shader/generator/glsl_shader_gen.cpp:67]: Can you leave a comment about the reason, something to the effect of: "Apple Silicon GPU drivers optimize more aggressively, which can create too much variance and cause visual artifacting in games like Pokemon."

**Comment by Steveice10** [comment]
(also just for future reference: this happens in the old OpenGL builds on Apple Silicon Macs as well, so it's not related to MoltenVK but the underlying Apple driver optimizations)

**Comment by GPUCode** [comment]
Insta-merging, cause it's a simple change

**Pull request merged**